File: go/muse/algorithms/sorting/sorting_test.go
package sorting

import (
	"testing"

	"muse/algorithms/sorting/tests"
)

func TestBubbleSort(t *testing.T) {
	t.Run("BubbleSort", tests.DeriveRandom(BubbleSort, 10000))
}

func TestInsertionSort(t *testing.T) {
	t.Run("InsertionSort", tests.DeriveRandom(InsertionSort, 10000))
}

func TestMergeSort(t *testing.T) {
	t.Run("MergeSort", tests.DeriveRandom(MergeSort, 100000))
}

func TestQuickSort(t *testing.T) {
	t.Run("QuickSort", tests.DeriveRandom(QuickSort, 100000))
}

func TestSelectionSort(t *testing.T) {
	t.Run("SelectionSort", tests.DeriveRandom(SelectionSort, 10000))
}

func TestShellSort(t *testing.T) {
	t.Run("ShellSort", tests.DeriveRandom(ShellSort, 100000))
}

File: python/muse/algorithms/sorting/sorting_test.py
from muse.algorithms import sorting
from muse.algorithms.sorting import tests
from muse.util import test_runner


def test_bubble_sort() -> bool:
    return tests.derive_random(sorting.bubblesort, 10000)()


def test_insertion_sort() -> bool:
    return tests.derive_random(sorting.insertionsort, 10000)()


def test_merge_sort() -> bool:
    return tests.derive_random(sorting.mergesort, 100000)()


def test_quick_sort() -> bool:
    return tests.derive_random(sorting.quicksort, 100000)()


def test_selection_sort() -> bool:
    return tests.derive_random(sorting.selectionsort, 10000)()


def main() -> None:
    test_runner.pick(test_bubble_sort)

    test_runner.pick(test_insertion_sort)

    test_runner.pick(test_merge_sort)

    test_runner.pick(test_quick_sort)

    test_runner.pick(test_selection_sort)


if __name__ == "__main__":
    main()
